client/state/jetpack-connect/reducer.js

```diff
@@ -126,6 +127,18 @@ export function jetpackConnectSite( state = {}, action ) {
 	return state;
 }
 
+export function jetpackConnectSitesList( state = {}, action ) {
```

Comment:
Do we need this under a new reducer tree? The `newSite` would make perfect sense in `jetpackConnectAuthorize` IMO.